File: varint.js
const fs = require('node:fs');

function readFile(filename) {
  const data = fs.readFileSync(filename);
  return Number(data.readBigInt64BE());
}

function encode(number) { 
  const buffer = [];

  while (number > 0x80) {
    buffer.push((number & 0x7F) | 0x80);
    number >>= 7;
  }

  buffer.push(number & 0x7F);

  return Buffer.from(buffer);
}

console.log(encode(readFile('150.uint64')));

function decode() {}


0x7F // 01111111 
150  // 10010110 -> 0000 0001
128  // 10000000

File: varint.py
import struct

def encode(n):
  out = []
  while n > 0:
    part = n % 128
    n >>= 7
    if n > 0:
      part |= 0x80
    out.append(part)
  return bytes(out)

with open('150.uint64', 'rb') as file:
  n = struct.unpack('>Q', file.read())[0]
  print(encode(n))


# 0x7F -> 01111111 
# 150  -> 10010110
# 128  -> 10000000